### PR TITLE
fix: catch-all paths replace double underscore with single underscore

### PR DESCRIPTION
**Pull request by mustafasegf**
close https://github.com/tuono-labs/tuono/issues/226

replace double underscore with single underscore

### CODE REVIEW

File: crates/tuono/src/route.rs [hunk 1]
```diff
@@ -261,4 +262,12 @@ mod tests {
             assert_eq!(route.output_file_path(), PathBuf::from(html))
         }
     }
+
+    #[test]
+    fn should_correctly_replace_double_underscore() {
+        let dyn_info = AxumInfo::new(&Route::new("/[posts__id]".to_string()));
```
Comment (Valerioageno) [inline crates/tuono/src/route.rs:268]: What about also testing the scenario with `dyn_` + `_catch_all` that is more similar to the real bug?

Comment (Valerioageno) [inline crates/tuono/src/route.rs:268]: I see a problem here. The user can define both `post__id` and `post_id` route, and they resolve with the same name.

How about replacing the `__` with the `u_` string?

Comment (Valerioageno) [reply crates/tuono/src/route.rs:268]: I think the best solution would be to remove the `_` from the replacement prefix. `catch_all_`and `hyphen_`

Comment (mustafasegf) [reply crates/tuono/src/route.rs:268]: hmm yeah that sounds more reasonable. let me remove the underscore prefix on the replacement logic

Comment (mustafasegf) [reply crates/tuono/src/route.rs:268]: one small issue is the generated name will be a bit harder to read for hypen
`"/posts/handle-this/[post]" => "posts_handlehyphen_this_dyn_post"`
the case I see where double underscore will happen on hypen if it's use at thestartd of route name in dynamic route or catch all route. where catch all does need dyn for it to work. i think it's fine to only change the `_catch_all_` to `catch_all_`

Comment (Valerioageno) [reply crates/tuono/src/route.rs:268]: It sounds the best. Agree

**Review by Valerioageno — APPROVED** [review]
LGTM. Thanks

**Pull request merged**